environment/src/main/java/com/sequenceiq/environment/environment/validation/securitygroup/aws/AwsEnvironmentSecurityGroupValidator.java

```diff
@@ -76,19 +79,27 @@ private void checkSecurityGroupVpc(EnvironmentDto environmentDto, ValidationResu
         CloudSecurityGroups securityGroups = platformParameterService.getSecurityGroups(request);
 
         boolean securityGroupInVpc = false;
-        for (CloudSecurityGroup cloudSecurityGroup : securityGroups.getCloudSecurityGroupsResponses().get(region.getName())) {
-            Object vpcId = cloudSecurityGroup.getProperties().get("vpcId");
-            if (cloudSecurityGroup.getGroupId().equals(securityGroupId)) {
-                LOGGER.info("Security group {} was found on AWS side.", securityGroupId);
-                if (vpcId != null && vpcId.toString().equals(environmentDto.getNetwork().getAws().getVpcId())) {
-                    securityGroupInVpc = true;
-                    break;
+        String awsVpcId = environmentDto.getNetwork().getAws().getVpcId();
+        Set<CloudSecurityGroup> cloudSecurityGroups = securityGroups.getCloudSecurityGroupsResponses().get(region.getName());
```

Comment:
This will yield an NPE if `securityGroups.getCloudSecurityGroupsResponses()` is `null`. I suggest wrapping this statement and `for` below inside a `if (Objects.nonNull(securityGroups.getCloudSecurityGroupsResponses())` condition.